### PR TITLE
upgrade/upgrades: make schema telemetry migration idempotent

### LINKED ISSUE #88273: upgrades: schema telemetry upgrade is not idempotent
**Describe the problem**

If the schedule already exists, we'll return an error, preventing a clean upgrade.

Jira issue: CRDB-19759

### PR DESCRIPTION
**Pull request by ajwerner**
Fixes #88273

Release note: None

### CODE REVIEW

**Review by postamar — APPROVED** [review]

**Review by Xiang-Gu — APPROVED** [review]
just nits

File: pkg/upgrade/upgrades/ensure_sql_schema_telemetry_schedule_test.go [hunk 1]
```diff
@@ -13,79 +13,103 @@ package upgrades_test
...
 	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
 
-	ctx := context.Background()
-	var args base.TestServerArgs
-	args.Knobs.JobsTestingKnobs = &jobs.TestingKnobs{
-		JobSchedulerEnv: jobstest.NewJobSchedulerTestEnv(
+	// We want to ensure that the migration will succeed when run again.
+	// To ensure that it will, we inject a failure when trying to mark
+	// the migration as complete when forceRetry is true.
```
Comment (Xiang-Gu) [inline pkg/upgrade/upgrades/ensure_sql_schema_telemetry_schedule_test.go:45]: Do you think reword "migration" to "upgrade" is better in the comments?

File: pkg/upgrade/upgrades/ensure_sql_schema_telemetry_schedule_test.go [hunk 2]
```diff
@@ -13,79 +13,103 @@ package upgrades_test
...
-	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{ServerArgs: args})
-	defer tc.Stopper().Stop(ctx)
-	tdb := sqlutils.MakeSQLRunner(tc.ServerConn(0))
+		)
+		args.Knobs.JobsTestingKnobs = jobKnobs
+		args.Knobs.SQLExecutor = &sql.ExecutorTestingKnobs{
+			BeforePrepare: func(ctx context.Context, stmt string, txn *kv.Txn) error {
+				if forceRetry && !injectedFailure.Get() && completeRegexp.MatchString(stmt) {
+					injectedFailure.Set(true)
+					return errors.New("booom")
```
Comment (Xiang-Gu) [inline pkg/upgrade/upgrades/ensure_sql_schema_telemetry_schedule_test.go:66]: nit: boom?

**Comment by ajwerner** [comment]
Done. TFTR!

bors r+

**Pull request merged**